File: VB/CustomWizardExample/Wizard/Presenters/ChoosePersonsDataSourceTypePage.vb
Imports System
Imports DevExpress.DataAccess.Wizard
Imports DevExpress.DataAccess.Wizard.Presenters
Imports DevExpress.XtraReports.Wizards3.Presenters

Namespace CustomWizardExample.Wizard.Presenters
	Public Class ChoosePersonsDataSourceTypePage
        Inherits WizardPageBase(Of Views.IChoosePersonsDataSourcePageView, Model.CustomReportModel)

        Public Sub New(ByVal view As Views.IChoosePersonsDataSourcePageView)
            MyBase.New(view)
        End Sub

		Public Overrides Sub Begin()
			View.DataSourceType = Model.PersonsDataSourceType
		End Sub

		Public Overrides Function Validate(<System.Runtime.InteropServices.Out()> ByRef errorMessage As String) As Boolean
			errorMessage = String.Empty
			If View.DataSourceType = PersonsDataSourceType.Static Then
				Return View.ShowWarning()
			End If
			Return True
		End Function


		Public Overrides Sub Commit()
			Model.PersonsDataSourceType = View.DataSourceType
			Model.DataSchema = New Person()
		End Sub

		Public Overrides Function GetNextPageType() As Type
            Return GetType(SelectColumnsPage(Of Model.CustomReportModel))
		End Function
		Public Overrides ReadOnly Property MoveNextEnabled() As Boolean
			Get
				Return True
			End Get
		End Property

	End Class
End Namespace

File: CS/CustomWizardExample/Wizard/Presenters/ChoosePersonsDataSourceTypePage.cs
using System;
using DevExpress.Data.WizardFramework;
using DevExpress.DataAccess.Wizard;
using DevExpress.DataAccess.Wizard.Presenters;
using DevExpress.XtraReports.Wizards3;
using DevExpress.XtraReports.Wizards3.Presenters;

namespace CustomWizardExample.Wizard.Presenters {
    public class ChoosePersonsDataSourceTypePage : WizardPageBase<Views.IChoosePersonsDataSourcePageView, XtraReportModel> {
        public ChoosePersonsDataSourceTypePage(Views.IChoosePersonsDataSourcePageView view) : base(view) { }

        public override void Begin() {
            View.DataSourceType = PersonsDataSourceType.Xml;
        }

        public override bool Validate(out string errorMessage) {
            errorMessage = string.Empty;
            if(View.DataSourceType == PersonsDataSourceType.Static)
                return View.ShowWarning();
            return true;
        }


        public override void Commit() {
            Model.Tag = View.DataSourceType;
            Model.DataSchema = new Person();
        }

        public override Type GetNextPageType() { return typeof(SelectColumnsPage<XtraReportModel>); }
        public override bool MoveNextEnabled { get { return true; } }
       
    }
}
